api/client/usermanager/client_test.go

```diff
@@ -4,170 +4,281 @@
 	// Delete the user.
-	err = s.usermanager.RemoveUser(tag.Name())
+	err = client.RemoveUser(tag.Name())
 	c.Assert(err, jc.ErrorIsNil)
-
-	// Assert that the user is gone.
-	_, err = s.State.User(tag)
-	c.Assert(err, gc.ErrorMatches, `user "jjam" is permanently deleted`)
-
-	err = user.Refresh()
-	c.Check(err, jc.ErrorIsNil)
-	c.Assert(user.IsDeleted(), jc.IsTrue)
 }
 
 func (s *usermanagerSuite) TestDisableUser(c *gc.C) {
-	user := s.Factory.MakeUser(c, &factory.UserParams{Name: "foobar"})
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
 
-	err := s.usermanager.DisableUser(user.Name())
-	c.Assert(err, jc.ErrorIsNil)
+	user := names.NewUserTag("foobar")
+	args := params.Entities{
+		[]params.Entity{{"user-foobar"}},
+	}
+	result := new(params.ErrorResults)
+	results := params.ErrorResults{
+		Results: make([]params.ErrorResult, 1),
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("DisableUser", args, result).SetArg(2, results).Return(nil)
 
-	err = user.Refresh()
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	err := client.DisableUser(user.Name())
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(user.IsDisabled(), jc.IsTrue)
 }
 
 func (s *usermanagerSuite) TestDisableUserBadName(c *gc.C) {
-	err := s.usermanager.DisableUser("not!good")
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	err := client.DisableUser("not!good")
 	c.Assert(err, gc.ErrorMatches, `"not!good" is not a valid username`)
 }
 
 func (s *usermanagerSuite) TestEnableUser(c *gc.C) {
-	user := s.Factory.MakeUser(c, &factory.UserParams{Name: "foobar", Disabled: true})
-
-	err := s.usermanager.EnableUser(user.Name())
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	user := names.NewUserTag("foobar")
+	args := params.Entities{Entities: []params.Entity{{user.String()}}}
+	result := new(params.ErrorResults)
+	results := params.ErrorResults{Results: make([]params.ErrorResult, 1)}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("EnableUser", args, result).SetArg(2, results).Return(nil)
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	err := client.EnableUser(user.Name())
 	c.Assert(err, jc.ErrorIsNil)
-
-	err = user.Refresh()
-	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(user.IsDisabled(), jc.IsFalse)
 }
 
 func (s *usermanagerSuite) TestEnableUserBadName(c *gc.C) {
-	err := s.usermanager.EnableUser("not!good")
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	client := usermanager.NewClientFromCaller(mockFacadeCaller)
+	err := client.EnableUser("not!good")
 	c.Assert(err, gc.ErrorMatches, `"not!good" is not a valid username`)
 }
 
 func (s *usermanagerSuite) TestCantRemoveAdminUser(c *gc.C) {
```

Comment:
This test was testing behavior outside of the api package. It can be removed. Verify that it is tested in state or the facade wherever the check is done.

Reply:
Confirmed It is tested in the facade.

Reply:
Removed it. Thanks.